charts/opentelemetry-operator/templates/deployment.yaml

```diff
@@ -49,6 +49,8 @@ spec:
             {{- end }}
             {{- if and .Values.manager.collectorImage.repository .Values.manager.collectorImage.tag }}
             - --collector-image={{ .Values.manager.collectorImage.repository }}:{{ .Values.manager.collectorImage.tag }}
+            {{- else if .Values.manager.collectorImage.repository }}
+            - --collector-image={{ .Values.manager.collectorImage.repository }}
```

Comment:
@jdiegosierra in the situation that we don't pass in a tag, are you sure the tag that ends up being uses is the default collector version from the operator and not the latest version of the set image repository?

Reply:
I tried with the latest version, so let me check it with a different one

Reply:
@TylerHelmuth It works as expected:
`appVersion: 0.100.1 `
```yaml
manager:
  image:
    repository: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator
    tag: "0.99.0"
  collectorImage:
    repository: ""
    tag: ""
```
```bash
k describe pod opentelemetry-operator-79cbd68444-wdpl9 | grep Image: 
    Image:         ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.99.0
```
```bash
k describe pod simplest-collector-54bb659675-99d7g | grep Image:
    Image:         ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector:0.99.0
```

Reply:
What about when you set and image repository and not a tag?

Reply:
In that scenario, the operator assumes the collector image is the latest one. If we want to have both components with the same version in that specific scenario, I guess we can pick the version from the operator.

```yaml
# Source: opentelemetry-operator/templates/deployment.yaml
  - --collector-image=otel/opentelemetry-collector
image: "ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.99.0"
```

Reply:
Anyway, that situation can not happen right now because it is mandatory to have both fields set:
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/ef0e1ac4f645cdbb9bd0108c76b1ed69e418430c/charts/opentelemetry-operator/templates/deployment.yaml#L50